package/MDAnalysis/lib/formats/include/readdcd.h

```diff
@@ -0,0 +1,783 @@
+  if (rc == -1) return DCD_BADEOF;	
+  return DCD_SUCCESS;
+}
+
+#define NFILE_POS 8L
+#define NSTEP_POS 20L
+
+static int write_dcdstep(fio_fd fd, int curframe, int curstep, int N, 
+			 const float *X, const float *Y, const float *Z, 
+			 const double *unitcell, int charmm) {
+  int out_integer;
+
+  if (charmm) {
+    /* write out optional unit cell */
+    if (unitcell != NULL) {
+      out_integer = 48; /* 48 bytes (6 doubles) */
+      fio_write_int32(fd, out_integer);
+      WRITE(fd, unitcell, out_integer);
+      fio_write_int32(fd, out_integer);
+    }
+  }
+
+  /* write out coordinates */
+  out_integer = N*4; /* N*4 bytes per X/Y/Z array (N floats per array) */
+  fio_write_int32(fd, out_integer);
+  WRITE(fd, X, out_integer);
+  fio_write_int32(fd, out_integer);
+  fio_write_int32(fd, out_integer);
+  WRITE(fd, Y, out_integer);
+  fio_write_int32(fd, out_integer);
+  fio_write_int32(fd, out_integer);
+  WRITE(fd, Z, out_integer);
+  fio_write_int32(fd, out_integer);
+
+  /* update the DCD header information */
+  fio_fseek(fd, NFILE_POS, FIO_SEEK_SET);
+  fio_write_int32(fd, curframe);
+  fio_fseek(fd, NSTEP_POS, FIO_SEEK_SET);
+  fio_write_int32(fd, curstep);
+  fio_fseek(fd, 0, FIO_SEEK_END);
+
+  return DCD_SUCCESS;
+}
+
+static int write_dcdheader(fio_fd fd, const char *remarks, int N, 
+			   int ISTART, int NSAVC, double DELTA, int with_unitcell,
+			   int charmm) {
+  int out_integer;
+  float out_float;
+  char title_string[241];
+  time_t cur_time;
+  struct tm *tmbuf;
+
+  out_integer = 84;
+  WRITE(fd, (char *) & out_integer, sizeof(int));
+  strcpy(title_string, "CORD");
+  WRITE(fd, title_string, 4);
+  fio_write_int32(fd, 0);      /* Number of frames in file, none written yet   */
+  fio_write_int32(fd, ISTART); /* Starting timestep                            */
+  fio_write_int32(fd, NSAVC);  /* Timesteps between frames written to the file */
+  fio_write_int32(fd, 0);      /* Number of timesteps in simulation            */
+  fio_write_int32(fd, 0);      /* NAMD writes NSTEP or ISTART - NSAVC here?    */
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  if (charmm) {
+    out_float = DELTA;
+    WRITE(fd, (char *) &out_float, sizeof(float));
+    if (with_unitcell) {
+      fio_write_int32(fd, 1);
+    } else {
+      fio_write_int32(fd, 0);
+    }
+  } else {
+    WRITE(fd, (char *) &DELTA, sizeof(double));
+  }
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  fio_write_int32(fd, 0);
+  if (charmm) {
+    fio_write_int32(fd, 24); /* Pretend to be Charmm version 24 */
+  } else {
+    fio_write_int32(fd, 0);
+  }
+  fio_write_int32(fd, 84);
+  fio_write_int32(fd, 164);
+  fio_write_int32(fd, 3); /* the number of 80 character title strings */
+
+  strncpy(title_string, remarks, 240);
+  WRITE(fd, title_string, 240);
```

Comment:
I increased the number because the read-write-read test cycle for headers read from test DCD files in the MDA test suite was leading to truncation -- I wanted to be able to fully reproduce the content of a read-in DCD file when writing back out.

As for the portability--I don't know--the lack of a rigid DCD standard has obviously caused issues in this effort. I wanted to be able to fully reproduce input DCD remarks content when writing, and that seems to be the case so far.